frontend/src/slices/gameDataSlice.ts

```diff
@@ -17,11 +18,14 @@ export const gameDataSlice = createSlice({
   reducers: {
     addBit: (state) => {
       state.numBits += 1;
+      state.totalNumBits += 1;
     },
     addBits: (state, action) => {
       state.numBits += action.payload.additionalBits;
+      state.totalNumBits += action.payload.additionalBits;
     },
     sellData: (state) => {
+      state.totalNumBits = state.numBits;
```

Comment:
This line is actually causing a bug! Think about what should happen to your lifetime total number of bits whenever you sell your data.